py/objmodule.c

```diff
@@ -171,12 +171,43 @@ STATIC const mp_rom_map_elem_t mp_builtin_module_table[] = {
 };
 MP_DEFINE_CONST_MAP(mp_builtin_module_map, mp_builtin_module_table);
 
+STATIC const mp_rom_map_elem_t mp_builtin_extensible_module_table[] = {
+    // builtin modules declared with MP_REGISTER_EXTENSIBLE_MODULE()
+    MICROPY_REGISTERED_EXTENSIBLE_MODULES
+};
+MP_DEFINE_CONST_MAP(mp_builtin_extensible_module_map, mp_builtin_extensible_module_table);
+
 // Attempts to find (and initialise) a builtin, otherwise returns
 // MP_OBJ_NULL.
-mp_obj_t mp_module_get_builtin(qstr module_name) {
-    mp_map_elem_t *elem = mp_map_lookup((mp_map_t *)&mp_builtin_module_map, MP_OBJ_NEW_QSTR(module_name), MP_MAP_LOOKUP);
+mp_obj_t mp_module_get_builtin(qstr module_name, bool extensible) {
+    mp_map_elem_t *elem = mp_map_lookup((mp_map_t *)(extensible ? &mp_builtin_extensible_module_map : &mp_builtin_module_map), MP_OBJ_NEW_QSTR(module_name), MP_MAP_LOOKUP);
     if (!elem) {
-        return MP_OBJ_NULL;
+        #if MICROPY_PY_SYS
+        // Special case for sys, which isn't extensible but can always be
+        // imported with the alias `usys`.
+        if (module_name == MP_QSTR_usys) {
+            return MP_OBJ_FROM_PTR(&mp_module_sys);
+        }
+        #endif
+
+        if (extensible) {
```

Comment:
Should this be `!extensible`?  Otherwise won't the code below check things like `upyb`?

Reply:
The key is that it only looks at `mp_builtin_extensible_module_map` here. I've updated the comment to explain this better.